drv/sidecar-front-io/src/leds.rs

```diff
@@ -0,0 +1,356 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+use drv_i2c_api::I2cDevice;
+use drv_i2c_devices::pca9956b::{Error, LedErr, Pca9956B};
+
+pub struct Leds {
+    controllers: [Pca9956B; 2],
+}
+
+/// Default LED Current
+///
+/// This will get written into the PCA9956B IREFALL register. The goal is to
+/// make these LEDs look as close to Gimlet CEM attention LEDs as possible.
+/// As of build C, Gimlet is pulling 50mA through those LEDs. From the PCA9956B
+/// datasheet, the calculus is I = IREFALL/256 * (900mV/Rext) * 1/4. Rext (R47
+///  & R48 on QSFP Front IO) is a 1K, so a bit of math results in a desired
+/// IREF value of d222 (hDE).
+const DEFAULT_LED_CURRENT: u8 = 222;
+
+/// Default LED PWM
+///
+/// This can be used to adjust LED duty cycle. The math here is simple, just
+/// PWM/256.
+const DEFAULT_LED_PWM: u8 = 255;
+
+/// There are two LED controllers, each controlling the LEDs on either the left
+/// or right of the board.
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum LedController {
+    Left = 0,
+    Right = 1,
+}
+
+// The necessary information to control a given LED.
+#[derive(Copy, Clone)]
+struct LedLocation {
+    controller: LedController,
+    output: u8,
+}
+
+/// FullErrorSummary takes the errors reported by each individual PCA9956B and
+/// maps them to a by-transceiver-port representation for the masks
+#[derive(Copy, Clone, Default, PartialEq, Eq)]
+pub struct FullErrorSummary {
+    pub overtemp_left: bool,
+    pub overtemp_right: bool,
+    pub system_led_err: bool,
```

Comment:
hmmm, do we want this to be an `LedErr` instead, to avoid losing information?

Reply:
I agree. I actually had it that way at first but convinced myself last night that I didn't need that. Today Aaron disagrees with that choice.